### PR TITLE
introduces options for particular profiles

### PR DESCRIPTION
**Pull request by mateuszlichon**
Most important changes:
- adds school admin controller, view, user creator, user chooser
- sets actions of student and teacher profile to those which they need
- builds school admin profile basing on general admin options but limiting them to particular school
- adjusts messages to particular profile options
- solves some minor mistakes along the way

### CODE REVIEW

File: src/main/java/pl/schoolmanager/controller/TeacherViewController.java [hunk 1]
```diff
@@ -74,16 +98,137 @@ public String createSubjectPost(@Valid @ModelAttribute Subject subject, BindingR
...
+	// READ
+	@GetMapping("/viewSubject/{subjectId}")
+	public String viewSubject(Model m, @PathVariable long subjectId) {
+		HttpSession s = SessionManager.session();
+		Subject subject = this.subjectRepository.findOne(subjectId);
+		s.setAttribute("subject", subject);
+		return "teacher_view/show_subject";
+	}
+
+	// UPDATE
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/TeacherViewController.java]: i would remove all such comments while method name is already saying what it does

File: src/main/java/pl/schoolmanager/controller/TeacherViewController.java [hunk 2]
```diff
@@ -74,16 +98,137 @@ public String createSubjectPost(@Valid @ModelAttribute Subject subject, BindingR
...
+	//CREATE FOR
+		@GetMapping("/createMark/{studentId}")
+		public String createMark(@PathVariable long studentId, Model m) {
+/*			HttpSession s = SessionManager.session();
+			Subject thisSubject = (Subject) s.getAttribute("subject");
+			Student thisStudent = this.studentRepository.findOne(studentId);*/
+			Mark mark = new Mark();
+/*			mark.setSubject(thisSubject);
+			mark.setStudent(thisStudent);*/
+			m.addAttribute("mark", mark);
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/TeacherViewController.java:151]: main repository should not contain commented code

File: src/main/java/pl/schoolmanager/controller/StudentController.java [hunk 3]
```diff
@@ -82,13 +84,15 @@ public String newStudentFromUserPost(@Valid @ModelAttribute Student student, Bin
...
-		return "redirect:/studentView/division";
+		return "redirect:/studentView/";
 	}
 	
 	// Managing exisitng student role
 	@GetMapping("/userStudent")
 	public String StudentFromUser(Model m) {
 		m.addAttribute("student", new Student());
+		HttpSession s = SessionManager.session();
+		s.invalidate();
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/StudentController.java]: You invalidate session here. What if this session contains another valuable information?

File: src/main/java/pl/schoolmanager/controller/SchoolAdminController.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+		Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+		String username = ((org.springframework.security.core.userdetails.User) principal).getUsername();
+		return this.userRepo.findOneByUsername(username);
+	}
+
+	@ModelAttribute("availableSchools")
+	public List<School> availableSchools() {
+		List<School> availableSchools = this.schoolRepo.findAll();
+		return availableSchools;
+	}
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/SchoolAdminController.java]: method can be shorter: return schoolRepo.findAll();

File: src/main/java/pl/schoolmanager/controller/SchoolAdminController.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+			if (s.getUserRole().getId() == thisUserRole.getId()) {
+				thisSchoolAdmin = s;
+			}
+		}
+		HttpSession s = SessionManager.session();
+		s.setAttribute("thisSchoolAdmin", thisSchoolAdmin);
+		s.setAttribute("thisSchool", thisSchoolAdmin.getSchool());
+		return "redirect:/schoolAdminView/";
+	}
+
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/SchoolAdminController.java]: very long method

File: src/main/java/pl/schoolmanager/controller/SchoolAdminController.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+	@GetMapping("/userSchoolAdmin")
+	public String TeacherFromUser(Model m) {
+		m.addAttribute("schoolAdmin", new SchoolAdmin());
+		HttpSession s = SessionManager.session();
+		s.invalidate();
+		return "school_admin/user_school_admin";
+	}
+
+	@PostMapping("/userSchoolAdmin")
+	public String TeacherFromUserPost(@Valid @ModelAttribute SchoolAdmin schoolAdmin, BindingResult bindingResult, Model m) {
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/SchoolAdminController.java:101]: very long method

File: src/main/java/pl/schoolmanager/controller/SchoolAdminController.java [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+	private User getLoggedUser() {
+		Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+		String username = ((org.springframework.security.core.userdetails.User) principal).getUsername();
+		return this.userRepo.findOneByUsername(username);
+	}
+
+	@ModelAttribute("availableSchools")
+	public List<School> availableSchools() {
+		List<School> availableSchools = this.schoolRepo.findAll();
+		return availableSchools;
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/SchoolAdminController.java:136]: method can be shorter: return schoolRepo.findAll();

File: src/main/java/pl/schoolmanager/controller/TeacherViewController.java [hunk 8]
```diff
@@ -74,16 +98,137 @@ public String createSubjectPost(@Valid @ModelAttribute Subject subject, BindingR
...
+	public String viewSubject(Model m, @PathVariable long subjectId) {
+		HttpSession s = SessionManager.session();
+		Subject subject = this.subjectRepository.findOne(subjectId);
+		s.setAttribute("subject", subject);
+		return "teacher_view/show_subject";
+	}
+
+	// UPDATE
+	@GetMapping("/updateSubject/{subjectId}")
+	public String updateSubject(Model m, @PathVariable long subjectId) {
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/TeacherViewController.java:106]: i would remove all such comments while method name is already saying what it does

File: src/main/java/pl/schoolmanager/controller/SchoolAdminController.java [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+	}
+	// MESSAGES INFO
+	@ModelAttribute("countAllReceivedMessages")
+	public Integer countAllReceivedMessages(Long receiverId) {
+		return this.messageRepository.findAllByReceiverId(getLoggedUser().getId()).size();
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/SchoolAdminController.java:141]: jpa repository can have methods like countByReceiver, countByReceiverId which return int value

File: src/main/java/pl/schoolmanager/controller/StudentViewController.java [hunk 10]
```diff
@@ -19,6 +19,10 @@
 	@Autowired
 	private StudentRepository studentRepository;
 	
+	@GetMapping("/")
+	public String stidentHome(Model m) {
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/StudentViewController.java]: typo

File: src/main/java/pl/schoolmanager/controller/SchoolAdminViewController.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/schoolAdminView")
+public class SchoolAdminViewController {
+
+	@GetMapping("/")
```
Comment (przemyslawkonik) [inline src/main/java/pl/schoolmanager/controller/SchoolAdminViewController.java:11]: get mapping can be also ("")

**Review by mateuszlichon — COMMENTED** [review]
session invalidation at login to profile role changed

**Pull request merged**